src/main/java/org/snomed/snowstorm/core/data/services/CodeSystemService.java

```diff
@@ -522,12 +522,18 @@ public CodeSystemVersion findLatestVisibleVersion(String shortName) {
 		return null;
 	}
 
-	public List<CodeSystemVersion> findByImportDateBetweenRange(long upperBoundEpoch, long lowerBoundEpoch) {
+	public List<CodeSystemVersion> findByImportDateBetweenRange(String shortName, long upperBoundEpoch, long lowerBoundEpoch) {
 		SearchHits<CodeSystemVersion> query = elasticsearchOperations.search(
 				new NativeSearchQueryBuilder()
-						.withQuery(boolQuery().should(rangeQuery(CodeSystemVersion.Fields.IMPORT_DATE)
-								.lte(epochToImportDate(upperBoundEpoch))
-								.gte(epochToImportDate(lowerBoundEpoch))))
+						.withQuery(
+								boolQuery()
+										.must(
+												rangeQuery(CodeSystemVersion.Fields.IMPORT_DATE).lte(epochToImportDate(upperBoundEpoch)).gte(epochToImportDate(lowerBoundEpoch))
+										)
+										.must(
+												matchQuery(CodeSystemVersion.Fields.SHORT_NAME, shortName)
```

Comment:
Here is better to use termQuery as shot name field is mapped as keyword type.

Reply:
I have made this change.